google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java

```diff
@@ -38,6 +39,15 @@ default Dialect getDialect() {
     throw new UnsupportedOperationException("method should be overwritten");
   }
 
+  /**
+   * Returns the SQL dialect that is used by the database.
+   *
+   * @return the SQL dialect that is used by the database.
+   */
+  default DatabaseDialect getDatabaseDialect() {
```

Comment:
Do we really need this? I think it will be confusing for customers when to use which of these two methods (`getDialect()` and `getDatabaseDialect()`). Could we instead instruct users to use `Dialect.toProto()` instead when they need the proto version?